query/src/main/java/org/apache/kylin/query/relnode/OLAPLimitRel.java

```diff
@@ -82,7 +82,8 @@ public void implementOLAP(OLAPImplementor implementor) {
         // ignore limit after having clause
         // ignore limit after another limit, e.g. select A, count(*) from (select A,B from fact group by A,B limit 100) limit 10
         // ignore limit after outer aggregate, e.g. select count(1) from (select A,B from fact group by A,B ) limit 10
-        if (!context.afterHavingClauseFilter && !context.afterLimit && !context.afterOuterAggregate) {
+        // ignore limit after join
+        if (!context.afterHavingClauseFilter && !context.afterLimit && !context.afterOuterAggregate && !context.afterJoin) {
```

Comment:
My concern is, this change may make many normal queries (join with the limit) much inefficient.

Reply:
Good point, I am looking for better solution now.